From ba26174eac18bf971d63d99175c9c8f62424b815 Mon Sep 17 00:00:00 2001
From: Gustavo Solaira <gustavos@codeaurora.org>
Date: Tue, 21 Aug 2018 22:07:58 -0700
Subject: [PATCH] Enable compilation of eap-qti-kernel on msm-4.4

Some changes are needed in the existing code to support
the EAP add-on package for kernel.

Change-Id: I68732c61f609d5420aa2c30524f3f20b9b5fea08
Signed-off-by: Gustavo Solaira <gustavos@codeaurora.org>
---
 include/linux/mod_devicetable.h | 13 +++++++++++++
 include/linux/socket.h          |  4 +++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/include/linux/mod_devicetable.h b/include/linux/mod_devicetable.h
index f534982..f02eb5d 100644
--- a/include/linux/mod_devicetable.h
+++ b/include/linux/mod_devicetable.h
@@ -677,4 +677,17 @@ struct ulpi_device_id {
 	kernel_ulong_t driver_data;
 };
 
+#define MHI_NAME_SIZE 32
+
+/**
+ * struct mhi_device_id - MHI device identification
+ * @chan: MHI channel name
+ * @driver_data: driver data;
+ */
+
+struct mhi_device_id {
+	const char chan[MHI_NAME_SIZE];
+	kernel_ulong_t driver_data;
+};
+
 #endif /* LINUX_MOD_DEVICETABLE_H */
diff --git a/include/linux/socket.h b/include/linux/socket.h
index 5bf59c8..9d3b16e 100644
--- a/include/linux/socket.h
+++ b/include/linux/socket.h
@@ -200,7 +200,8 @@ struct ucred {
 #define AF_ALG		38	/* Algorithm sockets		*/
 #define AF_NFC		39	/* NFC sockets			*/
 #define AF_VSOCK	40	/* vSockets			*/
-#define AF_MAX		41	/* For now.. */
+#define AF_QIPCRTR	42	/* QTI IPC Router		*/
+#define AF_MAX		43	/* For now.. */
 
 /* Protocol families, same as address families. */
 #define PF_UNSPEC	AF_UNSPEC
@@ -246,6 +247,7 @@ struct ucred {
 #define PF_ALG		AF_ALG
 #define PF_NFC		AF_NFC
 #define PF_VSOCK	AF_VSOCK
+#define PF_QIPCRTR	AF_QIPCRTR
 #define PF_MAX		AF_MAX
 
 /* Maximum queue length specifiable by listen.  */
-- 
1.9.1

